src/Orleans/project.json

```diff
@@ -1,6 +1,7 @@
-{
+﻿{
   "dependencies": {
-    "Newtonsoft.Json": "7.0.1"
+    "Newtonsoft.Json": "7.0.1",
+    "System.Collections.Immutable": "1.1.36"
```

Comment:
I tend to have the same reservations here. I am not sure what the best solution is for this? Maybe add a new project which is a DLL to put extended serialization logic in? This would probably allow the Json.Net to be moved out as well?